benchmarks/tool/compile_benchmarks.dart

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env dart
+
+import 'dart:io' show exit, Platform, Process, Directory;
+
+import 'package:args/args.dart' show ArgParser;
+import 'package:path/path.dart' as path;
+import 'package:pool/pool.dart' show Pool;
+
+Future<void> main(List<String> args) async {
+  final argParser = ArgParser()
+    ..addOption('target', mandatory: false, defaultsTo: 'jit,aot,js')
+    ..addOption('jobs', abbr: 'j', mandatory: false);
+
+  final parsedArgs = argParser.parse(args);
+
+  var jobs = Platform.numberOfProcessors;
+  if (parsedArgs['jobs'] != null) {
+    jobs = int.parse(parsedArgs['jobs']!);
+  }
+
+  var targets = <Target>{};
+  for (final targetStr in parsedArgs['target'].split(',')) {
+    switch (targetStr) {
+      case 'jit':
+        targets.add(jitTarget);
+        break;
+
+      case 'aot':
+        targets.add(aotTarget);
+        break;
+
+      case 'js':
+        targets.add(jsTarget);
+        break;
+
+      default:
+        print(
+            'Unsupported target: $targetStr. Supported targets: jit, aot, js');
+        exit(1);
+    }
+  }
+
+  // Arg list is immutable, clone it
+  var sourceFiles = List.from(parsedArgs.rest);
+
+  if (sourceFiles.isEmpty) {
+    // Compile all files in bin/
+    final dir = Directory('bin');
+    for (final dirFile in dir.listSync(recursive: false)) {
+      sourceFiles.add(dirFile.path);
+    }
+  }
+
+  final commands = [];
+
+  for (final sourceFile in sourceFiles) {
+    for (final target in targets) {
+      commands.add(target.compileArgs(sourceFile));
+    }
+  }
+
+  final pool = Pool(jobs);
+
+  final stream = pool.forEach(commands, (a) async {
+    var command = a as List<String>;
```

Comment:
I think the code is typeable, something like this:

```
final commands = <List<String>>[];
...
final = pool.forEach<List<String>, ProcessResult>(commands, (List<String> command) {
  print(command.join(' '));
  return Process.run(...);
});
```
(the `async` isn't needed - since there's no await in here)

Reply:
Done.